baseplate/sidecars/live_data_watcher.py

```diff
@@ -32,6 +37,64 @@ def __init__(self, dest: str, owner: int, group: int, mode: int):
         self.group = group
         self.mode = mode
 
+    @staticmethod
+    def get_encrypted_data_from_s3(
+        bucket_name: str, file_key: str, region_name: str, sse_key: str
+    ) -> Optional[bytes]:
+        s3_client = boto3.client(
+            "s3",
+            region_name=region_name,
+        )
+        data = None
+        # The S3 data must be Server Side Encrypted and we should have the decryption key.
+        kwargs = {
+            "Bucket": bucket_name,
+            "Key": file_key,
+            "SSECustomerKey": sse_key,
+            "SSECustomerAlgorithm": "AES256",
+        }
+        try:
+            s3_object = s3_client.get_object(**kwargs)
+            # Returns bytes.
+            data = s3_object["Body"].read()
+        except ClientError as error:
+            logger.exception(
+                "Failed to load from S3. Received error code %s: %s",
+                error.response["Error"]["Code"],
+                error.response["Error"]["Message"],
+            )
+            return None
+        except ValueError as error:
+            logger.exception(error)
+            return None
+        return data
+
+    @staticmethod
+    def get_data_to_write(json_data: dict, data: Optional[bytes]) -> Optional[bytes]:
+        # Check if we have a JSON in a special format containing:
+        # data = {
+        #    "live_data_watcher_load_type": str
+        # }
+        # If the load type is 'S3', this format is an indication that we support
+        # downloading the contents of encrypted files uploaded to S3.
+        if json_data.get("live_data_watcher_load_type") == "S3":
+            try:
+                bucket_name = json_data["bucket_name"]
+                file_key = json_data["file_key"]
+                sse_key = json_data["sse_key"]
+                region_name = json_data["region_name"]
```

Comment:
Is there any way that we can surface which one of these calls is failing in the `except` block?

Reply:
logger.exception naturally will surface it. example, if it's the sse_key, it'd look like this:

ERROR:baseplate.sidecars.live_data_watcher:Improper configuration found while attempting to load live data from S3.\nTraceback (most recent call last):\n  File "/src/baseplate/sidecars/live_data_watcher.py", line 84, in get_data_to_write\n    sse_key = json_data["sse_key"]\nKeyError: \'sse_key\''